scripts/o3de/o3de/download.py

```diff
@@ -111,46 +111,54 @@ def download_o3de_object(object_name: str, default_folder_name: str, dest_path:
     origin_uri = downloadable_object_data['origin_uri']
     parsed_uri = urllib.parse.urlparse(origin_uri)
 
-    download_zip_result = utils.download_zip_file(parsed_uri, download_zip_path, force_overwrite, object_name, download_progress_callback)
-    if download_zip_result != 0:
-        return download_zip_result
-
-    if not validate_downloaded_zip_sha256(downloadable_object_data, download_zip_path, f'{object_type}.json'):
-        logger.error(f'Could not validate zip, deleting {download_zip_path}')
-        os.unlink(download_zip_path)
-        return 1
-
     if not dest_path:
         dest_path = manifest.get_registered(default_folder=default_folder_name)
         dest_path = pathlib.Path(dest_path).resolve()
         dest_path = dest_path / object_name
     else:
         dest_path = pathlib.Path(dest_path).resolve()
 
-    if not dest_path:
-        logger.error(f'Destination path cannot be empty.')
-        return 1
-    if dest_path.exists():
-        if not force_overwrite:
-            logger.error(f'Destination path {dest_path} already exists.')
+    # If we have a git link then we should clone to the given download path here otherwise download and extract the zip
+    git_tuple = utils.is_git_provider_uri(parsed_uri)
+    if git_tuple[0]:
+        clone_result = git_tuple[2](parsed_uri.geturl(), dest_path)
+        if clone_result:
+            logger.error(f'Could not clone {parsed_uri.geturl()}')
```

Comment:
I think this logic would be more readable if you didn't use tuples here, instead use function names that make it clear what is happening
```suggestion
    # If we have a git link then we should clone to the given download path here otherwise download and extract the zip
    if utils.is_git_provider_uri(parsed_uri):
        clone_result = utils.git_clone(parsed_uri.geturl(), dest_path)
        if clone_result != 0:
            logger.error(f'Could not clone {parsed_uri.geturl()}')
```